transport.go

```diff
@@ -148,6 +148,10 @@ func getTransport(opts TransportOptions, encoding encoding.Encoding, tracer open
 		return nil, err
 	}
 
+	if enc == encoding.Protobuf {
```

Comment:
This currently ties the protobuf encoding to grpc. Theres some that as a result of evaluating the peer list infers that host:port combinations should default to tchannel. Unfortunately, at point in code the information if the protocol was inferred or specified is lost. If this is a problem I can attempt to refactor the code to infer protocol based on encoding, but allow overrides.